x-pack/plugins/security/server/authentication/tokens.ts

```diff
@@ -59,9 +57,13 @@ export class Tokens {
         access_token: accessToken,
         refresh_token: refreshToken,
         authentication: authenticationInfo,
-      } = await this.options.client.callAsInternalUser('shield.getAccessToken', {
-        body: { grant_type: 'refresh_token', refresh_token: existingRefreshToken },
-      });
+      } = (
+        await this.options.client.security.getToken<{
+          access_token: string;
```

Comment:
**question** Do you know if there type improvements we can eventually upstream to the ES client, or is this intentionally generic? I know there are varied responses from ES that we can get here, but there appers to be some common fields that we could potentially leverage.

Reply:
AFAIK we'll get [much better type definitions](https://github.com/elastic/elasticsearch-js/pull/1358/files#diff-9c72cd5b60f97485f79f0a568b27a60a117ac59fb8f3bf70b3dd612413e96155R9037) soon, see https://github.com/elastic/kibana/pull/83808 for more details.

Once this merges I think we can improve missing pieces on case-by-case basis.